### PR TITLE
Add joint filters info dict

### PR DESCRIPTION
**Pull request by KoalaQin**

### CODE REVIEW

**Comment by KoalaQin** [comment]
@mike-w-wilson Could we get this in first?

File: gnomad/utils/vcf.py [hunk 1]
```diff
@@ -1244,19 +1251,39 @@ def make_vcf_filter_dict(
...
-        },
-        "PASS": {"Description": "Passed all variant filters"},
-        variant_qc_filter: variant_qc_filter_dict[variant_qc_filter],
-    }
+    if joint:
+        filter_dict = {
+            "PASS": {
+                "Description": "Passed variant filters in exomes and genomes, "
+                "or Passed variant filters in one of the exomes "
+                "or genomes datasets but the variant was not present in the other dataset"
```
Comment (mike-w-wilson) [inline gnomad/utils/vcf.py:1259]:
```suggestion
                "Description": "Either passed all variant filters in both exomes and "
                "genomes, or passed all variant filters in either exomes or genomes "
                "while being absent from the other dataset"
```

File: gnomad/utils/vcf.py [hunk 2]
```diff
@@ -1244,19 +1251,44 @@ def make_vcf_filter_dict(
...
+            },
+            "BOTH_FILTERED": {
+                "Description": "Failed variant filters in both exomes and genomes datasets"
+            },
+        }
+    else:
+        # The cutoffs must be specified for the filter descriptions to be generated
+        if snp_cutoff is None or indel_cutoff is None or inbreeding_cutoff is None:
+            raise ValueError(
+                "snp_cutoff, indel_cutoff, and inbreeding_cutoff must be specified to generate filter descriptions."
```
Comment (mike-w-wilson) [inline gnomad/utils/vcf.py:1277]: Good catch! Let's just move it up to near the other ValueError and just add a `and not joint` to the condition

File: gnomad/utils/vcf.py [hunk 3]
```diff
@@ -1244,19 +1251,44 @@ def make_vcf_filter_dict(
...
+    if joint:
+        filter_dict = {
+            "PASS": {
+                "Description": "Passed variant filters in exomes and genomes, "
+                "or Passed variant filters in one of the exomes "
+                "or genomes datasets but the variant was not present in the other dataset"
+            },
+            "EXOMES_FILTERED": {
+                "Description": "Failed variant filters in exomes "
+                "dataset, but either passed variant filters in genomes dataset or the variant was not present in the genomes dataset"
```
Comment (mike-w-wilson) [inline gnomad/utils/vcf.py:1263]:
```suggestion
                "Description": "Failed variant filters in the exomes dataset and either "
                "passed all variant filters in the genomes dataset or the variant was "
                "not present in the genomes dataset"
```

File: gnomad/utils/vcf.py [hunk 4]
```diff
@@ -1244,19 +1251,44 @@ def make_vcf_filter_dict(
...
+            },
+            "GENOMES_FILTERED": {
+                "Description": "Failed variant filters in genomes "
+                "dataset, but either passed variant filters in exomes dataset or the variant was not present in the exomes dataset"
```
Comment (mike-w-wilson) [inline gnomad/utils/vcf.py:1267]:
```suggestion
                "Description": "Failed variant filters in the genomes dataset and either "
                "passed all variant filters in the exomes dataset or the variant was "
                "not present in the exomes dataset"
```

**Review by mike-w-wilson — CHANGES_REQUESTED** [review]

File: gnomad/utils/vcf.py [hunk 5]
```diff
@@ -1243,20 +1250,51 @@ def make_vcf_filter_dict(
...
+        filter_dict = {
+            "PASS": {
+                "Description": "Either passed all variant filters in both exomes and "
+                "genomes, or passed all variant filters in either exomes or genomes "
+                "while being absent from the other dataset"
+            },
+            "EXOMES_FILTERED": {
+                "Description": "Failed variant filters in the exomes dataset and either "
+                "passed all variant filters in the genomes dataset or the variant was "
+                "not present in the genomes dataset"
```
Comment (mike-w-wilson) [inline gnomad/utils/vcf.py:1273]:
```suggestion
                "Description": "Failed variant filters in the exomes dataset and either "
                "passed all variant filters in the genomes dataset or the variant was "
                "not present in the genomes dataset. Refer to 'exomes_filters'  within "
                "INFO for more information"
```

File: gnomad/utils/vcf.py [hunk 6]
```diff
@@ -1243,20 +1250,51 @@ def make_vcf_filter_dict(
...
+            },
+            "GENOMES_FILTERED": {
+                "Description": "Failed variant filters in the genomes dataset and either "
+                "passed all variant filters in the exomes dataset or the variant was "
+                "not present in the exomes dataset"
```
Comment (mike-w-wilson) [inline gnomad/utils/vcf.py:1278]:
```suggestion
                "Description": "Failed variant filters in the genomes dataset and either "
                "passed all variant filters in the exomes dataset or the variant was "
                "not present in the exomes dataset. Refer to 'genomes_filters' within "
                "INFO for more information"
```

File: gnomad/utils/vcf.py [hunk 7]
```diff
@@ -1243,20 +1250,51 @@ def make_vcf_filter_dict(
...
+            },
+            "BOTH_FILTERED": {
+                "Description": "Failed variant filters in both exomes and genomes datasets"
```
Comment (mike-w-wilson) [inline gnomad/utils/vcf.py:1281]:
```suggestion
                "Description": "Failed variant filters in both exomes and genomes datasets."
                "Refer to 'exomes_filters' and 'genomes_filters' within INFO for more information"
```

**Review by mike-w-wilson — CHANGES_REQUESTED** [review]
Forgot we want to reference the info filter fields

File: gnomad/utils/vcf.py [hunk 8]
```diff
@@ -1243,20 +1250,58 @@ def make_vcf_filter_dict(
...
+                "dataset or the variant was not present in the "
+                "genomes dataset. Refer to 'exomes_filters' within "
+                "INFO for more information"
+            },
+            "GENOMES_FILTERED": {
+                "Description": "Failed variant filters in the genomes dataset and "
+                "either passed all variant filters in the exomes "
+                "dataset or the variant was not present in the exomes "
+                "dataset. Refer to 'genomes_filters' within INFO for "
+                "more information"
```
Comment (mike-w-wilson) [inline gnomad/utils/vcf.py:1283]: Sorry why is this formatted into so many smaller lines instead of the regular 88 max? Did something change?

**Review by mike-w-wilson — COMMENTED** [review]
A question on the formatting happening

File: gnomad/utils/vcf.py (hunk 8 above)
Comment (KoalaQin) [reply gnomad/utils/vcf.py:1283]: I had 88 max for my coding style, but when I use pre-commit, it reformatted again....

Comment (KoalaQin) [reply gnomad/utils/vcf.py:1283]: Let me just copy yours.

Comment (mike-w-wilson) [reply gnomad/utils/vcf.py:1283]: I wonder if the new black is doing something strange then if its pre-commit formatting

Comment (KoalaQin) [reply gnomad/utils/vcf.py]: I think it's because of my hard-wrapping on 88 max coding style.

Comment (KoalaQin) [reply gnomad/utils/vcf.py]: Should look good now.

**Review by mike-w-wilson — APPROVED** [review]
LGTM

**Pull request merged**